definitions/procedures/puppet/disable_puppet.rb

```diff
@@ -0,0 +1,35 @@
+module Procedures::Puppet
+  class DisablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Disable Puppet feature'
+      confine do
+        feature(:puppet_server)
+      end
+    end
+
+    def run
+      execute!('foreman-rake db:migrate VERSION=0 SCOPE=foreman_puppet')
+      feature(:installer).run(installer_arguments_disabling_puppet.join(' '), :interactive => false)
+      packages_action(:remove, packages_to_remove, :assumeyes => true)
+    end
+
+    private
+
+    def installer_arguments_disabling_puppet
+      [
+        '--no-enable-puppet',
+        '--no-enable-foreman-plugin-puppet',
+        '--no-enable-foreman-cli-puppet',
```

Comment:
This will fail on a scenario where foreman isn't exposed, such as the `foreman-proxy-content` scenario. You need to account for that

Reply:
You mean the proxy scenario will fail? I hope I've accounted for that now hopefully.